pkg/database/database.go

```diff
@@ -97,3 +114,28 @@ func NewJSONHandle(aead encryption.AEAD) (*codec.JsonHandle, error) {
 
 	return h, nil
 }
+
+func createTriggers(ctx context.Context, sqlResourceClient *sdkcosmos.SQLResourcesClient, triggerResources []*sdkcosmos.SQLTriggerResource, resourceGroup, dbName, dbAccountName, location, containerName string) error {
```

Comment:
I believe this function and all its callers are the culprit of the failures in e2e and INT deployment.  This is called by NewXXX (monitors, database, subscriptions and billing), so each time those clients are created, the code sends PUT and GET of `Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/triggers`.  In fact, your first e2e from yesterday had 28 (14 PUTs, 14 GETs), hammering ARM.

Instead, what you should do is to have these triggers created by the arm template like those generated by `pkg/deploy`.

Reply:
So e2e passed in INT, however, I do agree, it is indeed hammering ARM, and I am aware of it. I think it is the polling stuff which is doing this.
Thanks for the suggestions, Let me give it more thought to come up with a solution for this.

Reply:
There are 5 trigger to be created so 5 PUT request is expected, I don't understand how come 7 PUT requests are sent.

Reply:
Okay, If we create these triggers through arm templates. Still these will be created when an RP deployment would be done. However, we directly run the RP when running locally and while running CI e2e tests.

Reply:
Sent you what ARM receives.  I don't really understand your last part, but like an infrastructure change, you don't test that kind of change on a local rp either.

Reply:
Okay, so implemented the ARM templates based solution. The code is pushed. Have a look.